### PR TITLE
[Gekidou] Groups API Calls

### PR DESCRIPTION
**Pull request by shazm**
#### Summary

This follows on the API modifications made in https://github.com/mattermost/mattermost-mobile/pull/6261, and adds the other missing methods required for https://github.com/mattermost/mattermost-mobile/pull/6276;

- Get memberships (users), teams and channels [relations] for a given Group
- Get groups associated to a given team, channel and user

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-43981

#### Checklist

- [ ] Added or updated unit tests (required for all new features)
- [ ] Has UI changes
- [ ] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
This PR was tested on: 

#### Release Note

```release-note

```

### CODE REVIEW

**Comment by enahum** [comment]
rename the actions to `fetch...`

**Review by larkox — APPROVED** [review]
Apart of Elias' comment, LGTM.

**Comment by shazm** [comment]
> rename the actions to `fetch...`

Fixed 👍

File: app/components/autocomplete/at_mention/at_mention.tsx [hunk 1]
```diff
@@ -172,8 +172,29 @@ const makeSections = (teamMembers: Array<UserProfile | UserModel>, usersInChanne
...
+
+const getFilteredChannelGroups = async (serverUrl: string, channelId: string, searchTerm: string) => {
+    const response = await fetchGroupsForChannel(serverUrl, channelId);
+
+    if (response && 'groups' in response) {
+        return response.groups.filter((g) => g.name.toLowerCase().includes(searchTerm.toLowerCase()));
+    }
+
+    return [];
+};
```
Comment (enahum) [inline app/components/autocomplete/at_mention/at_mention.tsx:193]: move to remote and rename to fetch

**Review by enahum — CHANGES_REQUESTED** [review]

**Review by enahum — APPROVED** [review]

**Pull request merged**